dotnet/src/Functions/Functions.UnitTests/OpenApi/OpenApiDocumentParserV30FeatureTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Plugins.OpenApi;
+using SemanticKernel.Functions.UnitTests.OpenApi.TestPlugins;
+using Xunit;
+
+namespace SemanticKernel.Functions.UnitTests.OpenApi;
+
+public class OpenApiDocumentParserV30FeatureTests
+{
+    /// <summary>
+    /// OpenAPI document stream.
+    /// </summary>
+    private readonly Stream _openApiDocument;
+
+    /// <summary>
+    /// System under test - an instance of OpenApiDocumentParser class.
+    /// </summary>
+    private readonly OpenApiDocumentParser _parser;
+
+    public OpenApiDocumentParserV30FeatureTests()
+    {
+        this._openApiDocument = ResourcePluginsProvider.LoadFromResource("openapi_feature_tests.json");
+        this._parser = new OpenApiDocumentParser();
+    }
+
+    [Fact]
+    public async Task ParsesAllOfAsync()
+    {
+        var spec = await this._parser.ParseAsync(this._openApiDocument);
+
+        Assert.NotEmpty(spec.Operations);
+        var op0 = spec.Operations.Single(static x => x.Path == "/fooBarAllOf" && x.Method == HttpMethod.Get);
```

Comment:
nit: just a tip – you can use the `operationId` property to associate an id with the operation and use it for the operation lookup.